erpnext/stock/doctype/material_request/material_request.js

```diff
@@ -339,50 +347,64 @@ frappe.ui.form.on("Material Request Item", {
 });
 
 erpnext.buying.MaterialRequestController = erpnext.buying.BuyingController.extend({
-	tc_name: function() {
+	tc_name: function () {
 		this.get_terms();
 	},
 
-	item_code: function() {
+	item_code: function (frm, cdt, cdn) {
 		// to override item code trigger from transaction.js
+		const row = locals[cdt][cdn];
+		if (row.item_code) {
+			frappe.call({
+				method: "erpnext.stock.doctype.material_request.material_request.check_compliance_item_category",
+				args: {
+					"item_code": row.item_code,
+					"item_category": "Dry Flower"
+				},
+				callback: function (readable) {
+					cur_frm.fields_dict.items.grid.set_column_disp("jar_type", readable.message);
+					cur_frm.fields_dict.items.grid.set_column_disp("unwind_direction", readable.message);
+				}
+			})
```

Comment:
We can actually use the existing API to do this, instead of a new server function

```suggestion
			frappe.db.get_value("Compliance Item", { "item_code": row.item_code }, "item_category")
			    .then(item => {
			    	if (item.message.item_category == "Dry Flower") {
						frm.fields_dict.items.grid.set_column_disp("jar_type", true);
						frm.fields_dict.items.grid.set_column_disp("unwind_direction", true);
					}
			    })
```